libs/mips-binutils/Makefile

```diff
@@ -1,22 +1,66 @@
-.PHONY: all clean
+.PHONY: install_libelf uninstall_libelf
+.PHONY: install_mips-binutils uninstall_mips-binutils
+.PHONY: install uninstall
+.PHONY: clean
 
-all: lib/libgtest.a
 
-lib/libgtest.a: lib/gtest-all.o
-	ar -rv lib/libgtest.a lib/gtest-all.o
+#########################################################################
+#                             MIPS_BINUTILS
+#########################################################################
 
-lib/gtest-all.o: lib/src/gtest-all.cc
-	g++ -I lib/include -I lib -c $< -o $@
+# Variables for convinience:
+PREFIX=/usr/local/mips-binutils
+ALIAS_DIR=/usr/local/bin
 
-lib/src/gtest-all.cc: release-1.8.0.tar.gz
-	rm -rf lib
-	tar xfm release-1.8.0.tar.gz
-	mv googletest-release-1.8.0/googletest lib
-	rm -rf googletest-release-1.8.0
+AS= $(PREFIX)/bin/mipsel-unknown-linux-gnu-as
+LD= $(PREFIX)/bin/mipsel-unknown-linux-gnu-ld
+OB= $(PREFIX)/bin/mipsel-unknown-linux-gnu-objdump
+MIPS_UTILS= $(AS) $(LD) $(OB)
 
-release-1.8.0.tar.gz:
-	wget https://github.com/google/googletest/archive/release-1.8.0.tar.gz
+ALIAS_AS= $(ALIAS_DIR)/mips-as
+ALIAS_LD= $(ALIAS_DIR)/mips-ld
+ALIAS_OB= $(ALIAS_DIR)/mips-objdump
+ALIASES= $(ALIAS_AS) $(ALIAS_LD) $(ALIAS_OBJ)
 
+DIR= binutils
+ARCHIVE= binutils-2.28.tar.bz2
+URL= http://ftp.gnu.org/gnu/$(DIR)/$(ARCHIVE)
```

Comment:
I don't like the way GoogleTest is included to the repository now (through wget). It is OK as we do not have anything better, but it is not good to copy that for other libraries.

The good thing about GoogleTest that it is not installed deeply inside your machine. Assume I am a external user who wants to try MIPT-MIPS. I played with it a little bit, I didn't like it and removed it completely by `rm -rf`. However, this installers did some implicit changes in Linux which may cause some unexpected consequences. For example, are you sure that installer would work correctly if previous version of libelf is installed? If _newer_ version is installed?

Another feature of GoogleTest installation is a real cross-platformness. I am not sure that your Makefiles would work on Solaris, BSD or something exotic, while manual can be easily adjusted by reader himself/herself.